### DISCUSSION TITLE
LFTP mirror equivalent in cadaver?

### DISCUSSION THREAD

**Discussion by ghost** [notroj/cadaver#18] [Q&A] [answered]
Hi, what would be the Cadaver equivalent of the following lftp command? 

mirror -R -vvv -n --no-perms --delete --exclude .git/

It takes all files from local host and transfers only those which are not already on the remote host. 
It excludes the folder .git/
It removes any file from the remote host that is not on the local host. 
It does not copy over any file permissions. 
It is a recursive command, so goes through all folders from the local directory. 

Thanks!
Andreas

**Comment by notroj** [comment, marked as answer]
Hi Andreas, there is no equivalent in cadaver. I built sitecopy for that kind of use case and it works with WebDAV.

**Comment by ghost** [comment]
Thank you! This seems like exactly what I need. 

EDIT: 

> I solved the below by:
> 
> 1) Copy any local changes to a safe place. 
> 2) sitecopy --fetch fm1107se
> 3) sitecopy --synch fm1107se
> 4) sitecopy --verify fm1107se (shows sites are identical)
> 5) Copy local changes back to my local site directory
> 6) sitecopy fm1107se ==> Shows the changes expected on the remote site
> 7) sitecopy --update fm1107se
> 
> The last command now works without the error. 

---- 

I tried the setup and the authentication works well, however when I run the command I get: 

```
sitecopy --update fm1107se
sitecopy: Updating site `fm1107se' (on webdav.fastmail.com in /USER.fastmail.com/files/1107/website/)
Uploading dar/pages/SimplexChat/index.html: [....] failed: 409 Conflict
sitecopy: Errors occurred while updating the remote site.
```
My config file:
```

site fm1107se
   server webdav.fastmail.com
   remote /USER.fastmail.com/files/1107/website/
   local ~/Documents/_projects/github/1107/
   username USER@fastmail.com
   password PASSWORD
   url https://webdav.fastmail.com/
   protocol webdav
   http secure
   exclude .git
```

Do you know if I'm missing something? Do I need to specify "force overwrite" somehow? I read the man-page and I couldn't find this option. 

Thank you
Andreas

**Reply by notroj** [reply]
The typical cases which produce a `409 Conflict` error are trying to replace a collection (directory) with a file or trying to create a new file in a directory which doesn't exist, so, yeah, a doing a `--fetch` first should fix them because it means the stored state doesn't match the actual remote state.
